content/blog/2023/nextflow-with-gbatch.md

```diff
@@ -0,0 +1,291 @@
+    input:
+    path transcriptome
+
+    output:
+    path 'salmon_index'
+
+    script:
+    """
+    salmon index --threads $task.cpus -t $transcriptome -i salmon_index
+    """
+}
+
+process QUANTIFICATION {
+    tag "Salmon on $sample_id"
+    publishDir params.outdir, mode:'copy'
+
+    input:
+    path salmon_index
+    tuple val(sample_id), path(reads)
+
+    output:
+    path "$sample_id"
+
+    script:
+    """
+    salmon quant --threads $task.cpus --libType=U -i $salmon_index -1 ${reads[0]} -2 ${reads[1]} -o $sample_id
+    """
+}
+
+process FASTQC {
+    tag "FASTQC on $sample_id"
+
+    input:
+    tuple val(sample_id), path(reads)
+
+    output:
+    path "fastqc_${sample_id}_logs"
+
+    script:
+    """
+    mkdir fastqc_${sample_id}_logs
+    fastqc -o fastqc_${sample_id}_logs -f fastq -q ${reads}
+    """
+}
+
+process MULTIQC {
+    publishDir params.outdir, mode:'copy'
+
+    input:
+    path '*'
+
+    output:
+    path 'multiqc_report.html'
+
+    script:
+    """
+    multiqc .
+    """
+}
+
+workflow {
+    Channel
+        .fromFilePairs(params.reads, checkIfExists: true)
+        .set { read_pairs_ch }
+
+    index_ch = INDEX(params.transcriptome_file)
+    quant_ch = QUANTIFICATION(index_ch, read_pairs_ch)
+    fastqc_ch = FASTQC(read_pairs_ch)
+    MULTIQC(quant_ch.mix(fastqc_ch).collect())
+}
+
+workflow.onComplete {
+    log.info ( workflow.success ? "\nDone! Open the following report in your browser --> $params.outdir/multiqc_report.html\n" : "Oops .. something went wrong" )
+}
+```
+
+> More information about this pipeline and where this came from can be found in the public training material developed by Seqera Labs at [https://training.seqera.io/](https://training.seqera.io/).
+
+#### Download the input data
+
+Create a folder named data in your project folder and download the example files (also taken from the Seqera training material).
+
+```bash
+$ mkdir data; cd data
+$ wget https://raw.githubusercontent.com/seqeralabs/nf-training-public/master/nf-training/data/ggal/gut_1.fq
+$ wget https://raw.githubusercontent.com/seqeralabs/nf-training-public/master/nf-training/data/ggal/gut_2.fq
+$ wget https://raw.githubusercontent.com/seqeralabs/nf-training-public/master/nf-training/data/ggal/transcriptome.fa
+$ cd ..
+```
+
+Files can also be uploaded directly to Google Cloud Storage, or downloaded directly from GitHub, depending on your own preferences.
+
+Use the command below to authenticate the Google Cloud Platform and direct Nextflow to use your account.
+
+```bash
+$ gcloud auth application-default login
+```
+
+#### Fire! Run the pipeline!
+
+With that done, you’re now ready to run the example Nextflow pipeline. Remember the name of the bucket you created? You will now use it to tell Nextflow here to move our input data and store intermediate and final data. That is, we're setting our work directory with the `-w` parameter.
+
+```bash
+$ nextflow run main.nf -w gs://rnaseq-pipeline-bckt/scratch
+```
```

Comment:
I suggest you specify the work dir in the config file instead, i.e. `workDir = ...`, to make the final command line as simple as possible.